### PR TITLE
Fixed blank dashboard page caused by null values

### LINKED ISSUE #687: Bug: Blank page when clicking on profile and coming back (not sure if duplicate)
### To Reproduce

1. Go to Home
2. Click on Login
3. (After getting logged in) Click on user profile picture on the top right.'
4. (After getting redirecting to github profile)  Click on the back button

### Expected behavior

Land back where the user left off before getting redirected

### Screenshots

image

### Desktop

- OS: Windows 10
- Browser: Google chrome
- Version:  83.0.4103.97 (Official Build) (64-bit)

### Smartphone

- Device:
- OS:
- Browser:
- Version:

### PR DESCRIPTION
**Pull request by jnielson94**
## What type of PR is this? (check all applicable)

- [ ] ♻️ Refactor
- [ ] ✨ Feature
- [x] 🐛 Bug Fix
- [ ] 👷 Optimization
- [ ] 📝 Documentation Update
- [ ] 🔖 Release
- [ ] 🚩 Other

## Description

Fix blank dashboard page issues by adding extra checks before property access.

## Related Tickets & Documents

Related to #617 and Fixes #687

## Mobile & Desktop Screenshots/Recordings (if there are UI changes)

No UI changes, unless you count the intended UI showing up as a change?

## Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [x] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 readme
- [x] 🙅 no documentation needed

### CODE REVIEW

**Comment by bdougie** [comment]
This is great @jnielson94. Thanks for jumping right in day one and contributing. I can confirm #687 is fixed, but also can confirm #617 is not. 

https://www.loom.com/share/f6eced3c023045b7b57a9d5ab1a0ac19

File: src/components/RepositoryGoals.js [hunk 1]
```diff
@@ -47,7 +47,7 @@ function RepositoryGoals({user}) {
     dispatch({type: "UPDATE", payload: updatedRepos});
   };
 
-  const data = repository.data && JSON.parse(repository.data.text);
+  const data = repository && repository.data && repository.data.text && JSON.parse(repository.data.text);
```
Comment (yg) [inline src/components/RepositoryGoals.js:50]: @jnielson94 I think one check is enough! WDYT?

```js
repository && repository.data && JSON.parse(repository.data.text);
```

Comment (yg) [reply src/components/RepositoryGoals.js:50]: Got it! `repository.data.text` is needed to check it is null or not!

**Comment by akshay5995** [comment]
@bdougie I still see we're using object && object.property && object.property.nestedProperty. Can we change Babel to add optional chaining?

**Comment by yg** [comment]
Yes, it would be great if we add support for **optional chaining**!

I & @jnielson94 chatted about this on Discord!

**Comment by akshay5995** [comment]
@yg Do we have an issue for that ? I'd like to pick it up

**Comment by yg** [comment]
> @yg Do we have an issue for that ? I'd like to pick it up

No @akshay5995! Feel free to create one issue/discussion and let us discuss there!

**Comment by bdougie** [comment]
@jnielson94 following up for here instead of Discord, just so the context is not lost.

> When loading the / dashboard, are we okay with a flash of the "you have no repository setup" screen? Or do we have a loading state that I can't figure out how to trigger while we check if they have a repository?

We are using [React Suspense to present a spinner](https://github.com/open-sauced/open-sauced/blob/678c445e3e337cb637a18308a05b6424540f8f51/src/containers/Dashboard.js#L9). A spinner is preferred.

>If you go to a repo you were tracking while logged out, are we okay that they see the homepage (which has a login button)? Or do we have a different page we'd expect them to land on? I've currently got it (locally) so that if they do login they end up on the page they want... I think?

Not sure if the scope of this fix can cover this, but I would love to provide the same experience as GitHub. Any repo is available to be seen, interactions require log in. 

Perhaps we can use the `isLoggedIn` attribute to show a login button on the repository page if the user is not logged in.

Screenshot 2020-06-20 06 33 35

**Comment by bdougie** [comment]
@jnielson94 I would love to merge this as-is to fix #687. Let us know if you are still on #617 and planning on addressing it in this PR.

**Comment by jnielson94** [comment]
@bdougie - I'm game to merge this as is. I adjusted the description so that #617 shouldn't be auto-closed when you do and I can make a new PR that works on fixing that one. 

I've done a little bit of digging on that one and the blank screen seems to be caused by this line throwing an error: https://github.com/open-sauced/open-sauced/blob/145eb1d1b8cd9b8bc0c51b430c2d4c653e3888cc/src/lib/reducers.js#L7

Since `state.repository` doesn't exist, and when you logged out the cached value for state ended up with an error related to no github auth existing, and `state.data.github` is null. Ideally the fix seems like it would involve avoiding the error in the cache when you log back in so that a spinner can show, but I haven't figured out how to clear the cache. (Will also post this comment on the issue itself)

**Pull request merged**